test/functional/feature_taproot.py

```diff
@@ -0,0 +1,594 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# Test Taproot softfork (BIPs 340-342)
+
+from test_framework.blocktools import create_coinbase, create_block, add_witness_commitment
+from test_framework.messages import CTransaction, CTxIn, CTxOut, COutPoint, CTxInWitness
+from test_framework.script import CScript, TaprootSignatureHash, taproot_construct, OP_0, OP_1, OP_CHECKSIG, OP_CHECKSIGVERIFY, OP_CHECKSIGADD, OP_IF, OP_CODESEPARATOR, OP_ELSE, OP_ENDIF, OP_DROP, LEAF_VERSION_TAPSCRIPT, SIGHASH_SINGLE, is_op_success, CScriptOp, OP_RETURN, OP_VERIF, OP_1NEGATE, OP_EQUAL, OP_SWAP, OP_CHECKMULTISIG, OP_CHECKMULTISIGVERIFY, OP_NOTIF, OP_2DROP, OP_NOT, OP_2DUP, OP_1SUB, OP_DUP, MAX_SCRIPT_ELEMENT_SIZE, LOCKTIME_THRESHOLD, ANNEX_TAG
```

Comment:
```suggestion
from test_framework.script import (
    ANNEX_TAG,
    CScript,
    CScriptOp,
    LEAF_VERSION_TAPSCRIPT,
    LOCKTIME_THRESHOLD,
    MAX_SCRIPT_ELEMENT_SIZE,
    OP_0,
    OP_1,
    OP_1SUB,
    OP_1NEGATE,
    OP_2DROP,
    OP_2DUP,
    OP_CHECKMULTISIG,
    OP_CHECKMULTISIGVERIFY,
    OP_CHECKSIG,
    OP_CHECKSIGADD,
    OP_CHECKSIGVERIFY,
    OP_CODESEPARATOR,
    OP_DROP,
    OP_DUP,
    OP_ELSE,
    OP_ENDIF,
    OP_EQUAL,
    OP_IF,
    OP_NOT,
    OP_NOTIF,
    OP_RETURN,
    OP_SWAP,
    OP_VERIF,
    SIGHASH_SINGLE,
    TaprootSignatureHash,
    is_op_success,
    taproot_construct,
)
```

Reply:
What a waste of lines. Is this more readable?

Reply:
Lines don't really cost anything. This is definitely *readable*, compared to the big blob it replaces.

Reply:
They cost screen space, and this isn't something that really requires reading.

Reply:
It feels like people can just scroll down if they aren't interested, or not if they are. I prefer to let the reader decide whether they desire to read or not.

Reply:
Done.